pageserver/src/walreceiver.rs

```diff
@@ -156,25 +107,18 @@ fn get_wal_producer_connstr(timelineid: ZTimelineId) -> String {
 //
 fn thread_main(
     conf: &'static PageServerConf,
-    timelineid: ZTimelineId,
     tenantid: ZTenantId,
-    interrupt_receiver: oneshot::Receiver<()>,
-) {
+    timelineid: ZTimelineId,
+) -> Result<()> {
```

Comment:
NIT: you don't really need to return the result here, and the closure that invokes this method can just wrap it in Ok(launch_wal_receiver(...)).

Might be more than one place that has this.

Reply:
I tried that now, but got:
```
  --> pageserver/src/walreceiver.rs:71:13
   |
71 |             thread_mgr::spawn(
   |             ^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `E` declared on the function `spawn`
```
This fixes it:
```
            thread_mgr::spawn(
                ThreadKind::WalReceiver,
                Some(tenantid),
                Some(timelineid),
                "WAL receiver thread",
                move || {
                    IS_WAL_RECEIVER.with(|c| c.set(true));
                    thread_main(conf, tenantid, timelineid);
                    Ok::<(), anyhow::Error>(())
                },
            )?;
```
But that ` Ok::<(), anyhow::Error>(())` looks pretty complicated, so I'm inclined to leave it as it is.